__tests__/pages/api/companies/[id]/employees/[employee_id]/archive_employee.test.ts

```diff
@@ -0,0 +1,183 @@
+import { v4 as uuidv4 } from 'uuid'
+import { prisma } from '@/lib/prisma'
+import handler from '@/pages/api/companies/[id]/employees/[employee_id]'
+import {
+  cleanDatabase,
+  mockDELETERequest,
+  parseJSON,
+} from '../../../../../../helpers'
+import type { Employee } from '@prisma/client'
+
+const ENDPOINT = '/api/companies/[id]/employees/[employee_id]'
+
+describe(`DELETE ${ENDPOINT}`, () => {
+  beforeEach(async () => {
+    await cleanDatabase(prisma)
+  })
+
+  describe('when company id is not a valid UUID', () => {
+    it('returns error', async () => {
+      const { req, res } = mockDELETERequest({ id: 'qwe' })
+
+      await handler(req, res)
+
+      expect(res._getStatusCode()).toBe(500)
+      expect(parseJSON(res)).toEqual({
+        success: false,
+        message: `Invalid UUID`,
+      })
+    })
+  })
+
+  describe('when the company does not exist', () => {
+    it('returns error', async () => {
+      const companyId = uuidv4()
+
+      const { req, res } = mockDELETERequest({ id: companyId })
+
+      await handler(req, res)
+
+      expect(res._getStatusCode()).toBe(404)
+      expect(parseJSON(res)).toEqual({
+        success: false,
+        message: `Company ${companyId} does not exist`,
+      })
+    })
+  })
+  describe('when company exists', () => {
+    describe('when the employee id is not a valid UUID', () => {
+      it('returns error', async () => {
+        const company = await prisma.company.create({
```

Comment:
We can move this code between 47 and 48 lines, because in this describe block (which told us "when company exists") we can cut off this company creation code into `beforeEach` block like this: https://github.com/optriment/web3-employment/blob/main/__tests__/pages/api/companies/%5Bid%5D/update_company.test.ts#L134